### ISSUE TITLE
Discrepancy between parse and playground trees

### ISSUE THREAD

**Issue by LinasVidziunas** [tree-sitter/tree-sitter#3449]
#### Description

First, I'd like to express my appreciation for this project. Thank you for your work on it.

I've noticed a discrepancy between the tree produced by the tree-sitter CLI's `parse` command and what is displayed in the playground for the [source code file](https://raw.githubusercontent.com/xrg/asterisk-xrg/10da3dab24e8ca08cf2c983f8d0206e383535b5a/channels/chan_iax2.c). Specifically:

1. The root node in the CLI `parse` output is `ERROR`.
2. The root node in the playground is `translation_unit`.
3. Overall, the playground seems to produce a more coherent tree structure.

#### Evidence

Please see the attached screenshots for a visual comparison:
<img width="1696" alt="Screenshot 2024-07-09 at 06 34 32" src="https://github.com/tree-sitter/tree-sitter/assets/17789443/bfc85439-bbd8-4b21-bef8-d114234fec23">
<img width="2047" alt="Screenshot 2024-07-09 at 06 36 06" src="https://github.com/tree-sitter/tree-sitter/assets/17789443/aa02b257-92f8-491e-afec-b8386bb3b22b">

#### Versions Tested

I've replicated this across multiple versions:

- [tree-sitter 0.22.2 macos arm64](https://github.com/tree-sitter/tree-sitter/releases/download/v0.22.2/tree-sitter-macos-arm64.gz)
- [tree-sitter 0.20.9 macos arm64](https://github.com/tree-sitter/tree-sitter/releases/download/v0.20.9/tree-sitter-macos-arm64.gz)

#### Questions

1. Can you provide any pointers into why the produced trees differ between the CLI and the playground?
2. Does the playground handle encountered errors more gracefully, leading to the observed differences?

Any clarification or guidance would be greatly appreciated. Thank you for your time.

### Steps to reproduce

```
### Setup grammar
git clone --depth=1 https://github.com/tree-sitter/tree-sitter-c/
cd tree-sitter-c/
npm install
tree-sitter generate
tree-sitter build --wasm

### Get the source code file
wget https://raw.githubusercontent.com/xrg/asterisk-xrg/10da3dab24e8ca08cf2c983f8d0206e383535b5a/channels/chan_iax2.c

### Generate tree using `parse`
tree-sitter parse chan_iax2.c --wasm --debug 2> log.txt 1> tree.txt

### Run playground
tree-sitter pg
# then paste the contents of the download file into the text area box
```

### Expected behavior

I would have expected the trees produced by the CLI `parse` command and the playground to be identical for the same input.

### Tree-sitter version (tree-sitter --version)

tree-sitter 0.22.6

### Operating system/version

macOS 14.5 (arm64)

**Comment by SunBK201** [comment]
I also encountered the same problem. Is this issue resolved?

**Comment by LinasVidziunas** [comment]
No, this issue is not resolved. I apologize for the confusion. I accidentally closed this issue while making some edits. Initially, I was using a C++ parser for a C file. I've now updated to the correct C parser. However, despite this change, I'm still encountering the same issue.

**Comment by WillLillis** [comment]
FWIW, this is most likely due to a version mismatch with the tree-sitter-c grammar, rather than tree-sitter itself. There is an issue open for adding version information to the playground (#2217), but I'm not sure how to check the versions currently.
